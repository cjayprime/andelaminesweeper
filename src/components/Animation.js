import React, {Component} from 'react';
import {Animated, Dimensions, Easing} from 'react-native';

import * as Style from '../assets/styles';

export default class Animation extends Component {
  constructor(props) {
    super(props);
    this.state = {
      translateY: new Animated.Value(Dimensions.get('window').height),
      scaleY: new Animated.Value(1),
    };
  }

  componentDidMount() {
    setTimeout(() => {
      this.pulsate();
    }, 100);
  }

  pulsate = () => {
    Animated.timing(this.state.translateY, {
      toValue: 0,
      duration: 1000,
      useNativeDriver: true,
      easing: Easing.easeOut,
    }).start(() => {
      const {scaleY} = this.state;
      Animated.sequence([
        Animated.timing(scaleY, {
          toValue: 1.4,
          duration: 125,
          useNativeDriver: true,
        }),
        Animated.timing(scaleY, {
          toValue: 1,
          duration: 125,
          useNativeDriver: true,
        }),
        Animated.timing(scaleY, {
          toValue: 1.4,
          duration: 125,
          useNativeDriver: true,
        }),
        Animated.timing(scaleY, {
          toValue: 1,
          duration: 125,
          useNativeDriver: true,
        }),
      ]).start();
    });
  };

  render() {
    const {children} = this.props;
    const {translateY, scaleY} = this.state;
    return (
      <Animated.View
        style={[
          Style.animation,
          {
            transform: [
              {
                translateY,
              },
              {
                scaleY,
              },
            ],
          },
        ]}>
        {children}
      </Animated.View>
    );
  }
}
