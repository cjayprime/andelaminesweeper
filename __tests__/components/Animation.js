import React from 'react';
import {View} from 'react-native';
import Animation from '../../src/components/Animation';

import renderer from 'react-test-renderer';

jest.useFakeTimers();
describe('Animation Test', () => {
    test('Stability', async () => {
        const tree = renderer.create(
            <Animation>
                <View/>
            </Animation>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    
        // jest.useFakeTimers();
        // setTimeout(() => {
        //      expect(component.state().fruits).toEqual(fruits);
        // }, 1500);
        // jest.runAllTimers();
    });
});