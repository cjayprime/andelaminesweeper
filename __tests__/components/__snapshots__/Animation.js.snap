// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animation Test Stability 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderRadius": 10,
      "elevation": 100,
      "left": "20%",
      "padding": 5,
      "position": "absolute",
      "top": "30%",
      "transform": Array [
        Object {
          "translateY": 1334,
        },
        Object {
          "scaleY": 1,
        },
      ],
      "width": "60%",
      "zIndex": 100000000000,
    }
  }
>
  <View />
</View>
`;
